src/net/tcp/socket.rs

```diff
@@ -82,6 +82,16 @@ impl TcpSocket {
         sys::tcp::set_reuseaddr(self.sys, reuseaddr)
     }
 
+    /// Sets the value of `SO_REUSEPORT` on this socket.
+    /// Only supported available in unix
+    #[cfg(all(
+        unix,
+        not(target_os = "solaris")
```

Comment:
I don't know if illumos has this.

/cc @tokio-rs/illumos

Reply:
Correct. It does not expose `SO_REUSEPORT`.
Including it in the negation with solaris would be wise for now.

Thanks for pointing that out.

Reply:
added illumos